drivers/gpu/drm/panel/panel-samsung-ea8061v-ams497ee01.c

```
@@ -0,0 +1,812 @@
+{
+	dsi_dcs_write_seq(ctx->dsi, 0xf7, 0x01);
+	return 0;
+}
+
+static int ea8061v_ams497ee01_update_aid(struct ea8061v_ams497ee01 *ctx,
+					 unsigned int brightness)
+{
+	int ret;
+	u8 aid_cmd[5] = { 0xb2, };
+
+	memcpy(aid_cmd + 1, seq_ea8061v_ams497ee01_aid +
+	       map_candela_to_aid[brightness], 4);
+
+	ret = mipi_dsi_dcs_write_buffer(ctx->dsi, aid_cmd,
+					ARRAY_SIZE(aid_cmd));
+	if (ret < 0)
+		return ret;
+	return 0;
+}
+
+static int ea8061v_ams497ee01_update_elvss(struct ea8061v_ams497ee01 *ctx,
+					   unsigned int brightness)
+{
+	int ret;
+	/* 0x5c for acl off, 0x4c for acl on */
+	u8 elvss_cmd[3] = { 0xb6, 0x5c };
+
+	memcpy(elvss_cmd + 2, ea8061v_ams497ee01_elvss_tbl +
+	       map_candela_to_elvss[brightness], 1);
+
+	ret = mipi_dsi_dcs_write_buffer(ctx->dsi, elvss_cmd,
+					ARRAY_SIZE(elvss_cmd));
+	if (ret < 0)
+		return ret;
+	return 0;
+}
+
+static int ea8061v_ams497ee01_update_gamma(struct ea8061v_ams497ee01 *ctx,
+					   unsigned int brightness)
+{
+	struct backlight_device *bl_dev = ctx->bl_dev;
+	u8 data[EA8061V_AMS497EE01_GAMMA_CMD_CNT + 1] = { 0xca, };
+	int ret;
+
+	memcpy(data + 1, seq_ea8061v_ams497ee01_lux + brightness,
+	       EA8061V_AMS497EE01_GAMMA_CMD_CNT);
+
+	ret = mipi_dsi_dcs_write_buffer(ctx->dsi, data, ARRAY_SIZE(data));
+	if (ret < 0)
+		return ret;
+
+	bl_dev->props.brightness = brightness;
+
+	return 0;
+}
+
+static int ea8061v_ams497ee01_set_brightness(struct backlight_device *bl_dev)
+{
+	struct ea8061v_ams497ee01 *ctx = bl_get_data(bl_dev);
+	unsigned int brightness = bl_dev->props.brightness;
+	struct mipi_dsi_device *dsi = ctx->dsi;
+
+	/* aid/aor */
+	ea8061v_ams497ee01_update_aid(ctx, brightness);
+	/* acl opr on */
+	dsi_dcs_write_seq(dsi, 0xb5, 0x29);
+	/* acl off (0x55, 0x02 for 15 % ACL) */
+	dsi_dcs_write_seq(dsi, 0x55, 0x00);
+	/* elvss b6 */
+	ea8061v_ams497ee01_update_elvss(ctx, brightness);
+
+	/* gamma ca */
+	ea8061v_ams497ee01_update_gamma(ctx, brightness);
+	ea8061v_ams497ee01_apply_gamma(ctx);
```

There is error checking missing for all the `ea8061v_*` functions. The `dsi_dcs_write_seq` one has magic error checking inbuilt, quite confusing sadly.

---

I dunno smart \what to do for this, check return value after each func call? maybe we can just ignore error ret an have more readable code..? :)